src/DocumentDB.ChangeFeedProcessor/PartitionManagement/RemainingWorkEstimator.cs

```diff
@@ -19,69 +22,73 @@ internal class RemainingWorkEstimator : IRemainingWorkEstimator
-                options.RequestContinuation = existingLease.ContinuationToken;
-                options.StartFromBeginning = string.IsNullOrEmpty(existingLease.ContinuationToken);
-
-                IChangeFeedDocumentQuery<Document> query = this.feedDocumentClient.CreateDocumentChangeFeedQuery(this.collectionSelfLink, options);
-                IFeedResponse<Document> response = null;
+                return new List<RemainingPartitionWork>().AsReadOnly();
+            }
 
-                try
+            var tasks = Partitioner.Create(leases)
+                .GetPartitions(this.degreeOfParallelism)
+                .Select(partition => Task.Run(async () =>
                 {
-                    response = await query.ExecuteNextAsync<Document>().ConfigureAwait(false);
-                    long parsedLSNFromSessionToken = TryConvertToNumber(ExtractLSNFromSessionToken(response.SessionToken));
-                    long lastQueryLSN = response.Count > 0 ?
-                        TryConvertToNumber(GetFirstDocument(response).GetPropertyValue<string>(LSNPropertyName)) - 1
-                        : parsedLSNFromSessionToken;
-                    if (lastQueryLSN < 0)
+                    IList<RemainingPartitionWork> results = new List<RemainingPartitionWork>();
+                    using (partition)
                     {
-                        // Could not parse LSN from document, we cannot determine the amount of changes but since the query returned 1 document, we know it's at least 1
-                        remainingWork += 1;
-                        continue;
+                        while (partition.MoveNext())
+                        {
+                            ILease item = partition.Current;
+                            try
+                            {
+                                if (string.IsNullOrEmpty(item.PartitionId)) continue;
+                                var result = await this.GetRemainingWorkForLeaseAsync(item);
+                                results.Add(new RemainingPartitionWork(item.PartitionId, result));
+                            }
+                            catch (DocumentClientException ex)
+                            {
+                                Logger.WarnException($"Getting estimated work for {item.PartitionId} failed!", ex);
+                            }
+                        }
                     }
 
-                    long partitionRemainingWork = parsedLSNFromSessionToken - lastQueryLSN;
-                    remainingWork += partitionRemainingWork < 0 ? 0 : partitionRemainingWork;
-                }
-                catch (DocumentClientException clientException)
-                {
-                    Logger.WarnException("GetEstimateWork > exception: partition '{0}'", clientException, existingLease.PartitionId);
-                }
-            }
+                    return results.ToArray();
+                })).ToArray();
```

Comment:
the problem is that in case the estimation work fails, I'd like to report at least a partial/uncomplete result. At least something. 

So, if I define 
var x = Partitioner.Create(leases).Select(partition => Task.Run(async () => { ... }));
then await Task.WhenAll(x); is OK

but then 

return tasks.Where(t => t.Status == TaskStatus.RanToCompletion).SelectMany(t => t.Result).ToList().AsReadOnly();

would be 0 -> because we are iterating from scratch all partitions. 

so I'd like to remember all tasks and wait for all of them to finish and then report the results from successfull runs.